### ISSUE TITLE
Can no copy text from Magic5Pro to Windows

### ISSUE THREAD

**Issue by wuerbuzuo** [Genymobile/scrcpy#3885]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: Windows10
 - scrcpy version: 2.0
 - installation method: Windows release
 - device model:Honor Magic5Pro
 - Android version: 13

**Describe the bug**
Terminal output when I press CTRL-C in scrcpy window:
```
[server] INFO: Device: HONOR PGT-AN10 (Android 13)
INFO: Renderer: direct3d
INFO: Initial texture: 1312x2848
[server] ERROR: Could not invoke method
java.lang.NoSuchMethodException: android.content.IClipboard$Stub$Proxy.getPrimaryClip [class java.lang.String, class java.lang.String, int]
        at java.lang.Class.getMethod(Class.java:2103)
        at java.lang.Class.getMethod(Class.java:1724)
        at com.genymobile.scrcpy.wrappers.ClipboardManager.getGetPrimaryClipMethod(ClipboardManager.java:34)
        at com.genymobile.scrcpy.wrappers.ClipboardManager.getText(ClipboardManager.java:82)
        at com.genymobile.scrcpy.Device.getClipboardText(Device.java:248)
        at com.genymobile.scrcpy.Device$2.dispatchPrimaryClipChanged(Device.java:109)
        at android.content.IOnPrimaryClipChangedListener$Stub.onTransact(IOnPrimaryClipChangedListener.java:84)
        at android.os.Binder.execTransactInternal(Binder.java:1323)
        at android.os.Binder.execTransact(Binder.java:1279)
[server] ERROR: Could not invoke method
java.lang.NoSuchMethodException: android.content.IClipboard$Stub$Proxy.getPrimaryClip [class java.lang.String, class java.lang.String, int]
        at java.lang.Class.getMethod(Class.java:2103)
        at java.lang.Class.getMethod(Class.java:1724)
        at com.genymobile.scrcpy.wrappers.ClipboardManager.getGetPrimaryClipMethod(ClipboardManager.java:34)
        at com.genymobile.scrcpy.wrappers.ClipboardManager.getText(ClipboardManager.java:82)
        at com.genymobile.scrcpy.Device.getClipboardText(Device.java:248)
        at com.genymobile.scrcpy.Device$2.dispatchPrimaryClipChanged(Device.java:109)
        at android.content.IOnPrimaryClipChangedListener$Stub.onTransact(IOnPrimaryClipChangedListener.java:84)
        at android.os.Binder.execTransactInternal(Binder.java:1323)
        at android.os.Binder.execTransact(Binder.java:1279)
```

**Comment by rom1v** [comment]
The vendor modified the method so iy is different from AOSP.

Please retrieve the framework library:

```
adb pull /data/system/framework.jar
```

and post it.

**Comment by wuerbuzuo** [comment]
> The vendor modified the method so iy is different from AOSP.
> 
> Please retrieve the framework library:
> 
> ```
> adb pull /data/system/framework.jar
> ```
> 
> and post it.

```
adb: error: failed to stat remote object '/data/system/framework.jar': No such file or directory
```

**Comment by rom1v** [comment]
Sorry, wrong path:

```
adb pull /system/framework/framework.jar
```

**Comment by wuerbuzuo** [comment]
> Sorry, wrong path:
> 
> ```
> adb pull /system/framework/framework.jar
> ```

FYI.
[magic5framework.zip](https://github.com/Genymobile/scrcpy/files/11128955/magic5framework.zip)

**Comment by rom1v** [comment]
They added a parameter `String stackTraceMsg` compared to the official Android:

```java
public ClipData getPrimaryClip(String pkg, int userId, String stackTraceMsg)
```

<details><summary>diff</summary>

```diff
diff --git a/server/src/main/java/com/genymobile/scrcpy/wrappers/ClipboardManager.java b/server/src/main/java/com/genymobile/scrcpy/wrappers/ClipboardManager.java
index cb176cc3a..7b7509755 100644
--- a/server/src/main/java/com/genymobile/scrcpy/wrappers/ClipboardManager.java
+++ b/server/src/main/java/com/genymobile/scrcpy/wrappers/ClipboardManager.java
@@ -37,8 +37,13 @@ public class ClipboardManager {
                         getPrimaryClipMethod = manager.getClass().getMethod("getPrimaryClip", String.class, String.class, int.class);
                         getMethodVersion = 1;
                     } catch (NoSuchMethodException e2) {
-                        getPrimaryClipMethod = manager.getClass().getMethod("getPrimaryClip", String.class, String.class, int.class, int.class);
-                        getMethodVersion = 2;
+                        try {
+                            getPrimaryClipMethod = manager.getClass().getMethod("getPrimaryClip", String.class, String.class, int.class, int.class);
+                            getMethodVersion = 2;
+                        } catch (NoSuchMethodException e3) {
+                            getPrimaryClipMethod = manager.getClass().getMethod("getPrimaryClip", String.class, int.class, String.class);
+                            getMethodVersion = 3;
+                        }
                     }
                 }
             }
@@ -80,8 +85,10 @@ public class ClipboardManager {
                 return (ClipData) method.invoke(manager, FakeContext.PACKAGE_NAME, FakeContext.ROOT_UID);
             case 1:
                 return (ClipData) method.invoke(manager, FakeContext.PACKAGE_NAME, null, FakeContext.ROOT_UID);
-            default:
+            case 2:
                 return (ClipData) method.invoke(manager, FakeContext.PACKAGE_NAME, null, FakeContext.ROOT_UID, 0);
+            default:
+                return (ClipData) method.invoke(manager, FakeContext.PACKAGE_NAME, FakeContext.ROOT_UID, null);
         }
     }
 
```

</details>

Please test this binary:

- [`scrcpy-win64-3885.zip`](https://tmp.rom1v.com/scrcpy/3885/1/scrcpy-win64-3885.zip) <sub>`SHA-256: 9bd2b32f6480319b8032651df8a7d637d5c66a407f43ddfc4241d91bd469810f`</sub>

**Comment by wuerbuzuo** [comment]
it works.Thank you so much!

**Comment by rom1v** [comment]
Merged into `dev`: f77e1c474edfafcd46f692452413f35a18c70949

It will be fixed in the next release.
